### PR TITLE
Computer speed

### PR DESCRIPTION
**Pull request by fabientownsend**

### CODE REVIEW

**Comment by felipesere** [comment]
I was hoping to see a test annotated with a timeout, to make sure you beat the time you need:
https://github.com/junit-team/junit4/wiki/timeout-for-tests

File: src/main/java/tictactoe/ComputerPlayer.java [hunk 1]
```diff
@@ -20,78 +22,114 @@ public final int nextMove() {
...
+            return bestPosition;
+        }
+    }
+
+    private boolean perfectMoveFound(int bestMoveEvaluated) {
+        return bestMoveEvaluated == 1;
+    }
+
+    private boolean couldEvaluateMove(int bestMoveEvaluated) {
+        return bestMoveEvaluated != -2;
```
Comment (felipesere) [inline src/main/java/tictactoe/ComputerPlayer.java]: This use of the `valueMove` variable feels odd.
Is it actually `theValueOfPossiblyTheBestMove`? Or Is it `theValueOfThePossiblyBestMoveUnlessItsTwoThenItsTheFactThatICouldNotAnswerFastEnough`?

If it is the second case, consider creating a small class, `ValuedMove` that captures the notion of "A move that has a value, but might also not have been evaluated" and has has meaningful methods and fields and tests... Micro-Types are OK. They are pretty cool. They give meaning and intent to simple things as "int or null" 😄

File: src/main/java/tictactoe/ComputerPlayer.java [hunk 2]
```diff
@@ -20,78 +22,114 @@ public final int nextMove() {
...
+    private final double LIMIT_SECONDS_FOUND_MOVE = 0.1;
+    private int alphaBetaPruning(Board board, Marks currentMark, int alpha, int beta, int depth, double startTime) {
         if (board.win(this.mark)) {
             return 1;
         } else if (board.win(oppositePlayer(this.mark))) {
             return -1;
         } else if (board.tie()) {
             return 0;
+        } else if (timeSpent(startTime) > LIMIT_SECONDS_FOUND_MOVE) {
+            return  -2;
```
Comment (felipesere) [inline src/main/java/tictactoe/ComputerPlayer.java]: See above...

File: src/main/java/tictactoe/ComputerPlayer.java (hunk 1 above)
Comment (fabientownsend) [reply src/main/java/tictactoe/ComputerPlayer.java]: It's more the second case, I definitely want to try with a Micro-Type!

File: src/test/java/tictactoe/GamePlayTest.java [hunk 3]
```diff
@@ -21,16 +25,47 @@ public void initialize() {
...
-        Board board = new Board();
+        Board board = new Board(3);
         Player player_1 = new ComputerPlayer(Marks.CROSS, board);
         Player player_2 = new ComputerPlayer(Marks.ROUND, board);
         this.game = new GamePlay(fakeCommandLine, board, player_1, player_2);
         game.play();
         assertTrue(out.toString().contains("tie"));
     }
 
+    @Test(timeout = 18000)
```
Comment (christophgockel) [inline src/test/java/tictactoe/GamePlayTest.java:38]: 👍 for a time constraint on the test level

**Pull request merged**